src/extensions/animation/index.js

```diff
@@ -108,89 +93,73 @@ function applyAnimationSettings( extraProps, blockType, attributes ) {
 	return extraProps;
 }
 
-addFilter(
-	'blocks.getSaveContent.extraProps',
-	'coblocks/applyAnimationSettings',
-	applyAnimationSettings
-);
-
-const enhance = compose(
-	/**
-	 * For blocks whose block type doesn't support `multiple`, provides the
-	 * wrapped component with `originalBlockClientId` -- a reference to the
-	 * first block of the same type in the content -- if and only if that
-	 * "original" block is not the current one. Thus, an inexisting
-	 * `originalBlockClientId` prop signals that the block is valid.
-	 *
-	 * @param {Function} WrappedBlockEdit A filtered BlockEdit instance.
-	 * @return {Function} Enhanced component with merged state data props.
-	 */
-	withSelect( ( select ) => {
-		return { selected: select( 'core/block-editor' ).getSelectedBlock(), select };
-	} )
-);
-
-const withAnimationSettings = createHigherOrderComponent( ( BlockListBlock ) => {
-	return enhance( ( { select, ...props } ) => {
-		let wrapperProps = props.wrapperProps;
-
-		const blockName	= select( 'core/block-editor' ).getBlockName( props.rootClientId || props.clientId );
-
-		// Block not supported return unmodified.
-		if ( ! allowedBlocks.some( ( allowedBlock ) => allowedBlock.blockType === blockName && ! allowedBlock.animateChildren ) ) {
-			return <BlockListBlock { ...props } />;
-		}
+/**
+ * React hook function to override the default block element to add wrapper props.
+ *
+ * @function useEditorProps
+ * @param {Object} block        Contains the selected block.
+ * @param {string} blockName    String referencing the selected block.
+ * @param {Object} props        Object with block props.
+ * @param {Object} wrapperProps Object with wrapper props used to extend selected block.
+ * @return {Object} Wrapper props to apply to block.
+ */
+const useEditorProps = ( block, blockName, props, wrapperProps ) => {
+	// Apply animation classes to block wrapper only if the animation attribute has changed.
+	const { animation } = props.attributes;
+	const prevAnimation = useRef();
 
-		if ( ! props.attributes?.animation ) {
-			// If animation is unset ensure that `coblocks-animate` class is unset.
-			if ( props.attributes?.className?.includes( 'coblocks-animate' ) ) {
-				const classList = props.attributes?.className.split( ' ' );
-				props.attributes.className = classList.filter( ( c ) => c !== 'coblocks-animate' ).join( ' ' );
-			}
+	// Hide the Block Toolbar while animations are playing.
+	const [ isAnimating, setIsAnimating ] = useState( false );
 
-			return <BlockListBlock { ...props } />;
+	const blockIsAllowed = allowedBlocks.some( ( allowedBlock ) => allowedBlock.blockType === blockName && ! allowedBlock.animateChildren );
+
+	useEffect( () => {
+		if ( !! blockIsAllowed && !! animation ) {
+			prevAnimation.current = animation;
 		}
+	}, [ animation ] );
```

Comment:
You don't want useEffect to rerun if blockIsAllowed is changed? If you do you need it in the passed props.

Reply:
The `blockIsAllowed` variable is based on the constant set on line 19. This constant does not receive mutations and therefore we do not expect the value to change after the initial computation of its value.